frontend/app/models/search.rb

```diff
@@ -10,12 +10,17 @@ def self.for_type(repo_id, type, criteria)
   end
 
   # :context_criteria - added to criteria for building the query, but hidden from user
+  # :sorting - ignore repo preferences and use provided sort for results (i.e. for typeaheads)
   def self.all(repo_id, criteria, context_criteria = {})
     context_criteria.each do |k, v|
-      raise "Not implemented" unless v.is_a? Array
-      criteria[k] ||= []
-      criteria[k] += v
```

Comment:
I'm not sure why the criteria value was restricted to Array before, but I've just added the ability for it to accept a non-Array as well. (If not, `SearchResultData` won't be able to handle the `sort` criteria)